libraries/Microsoft.Bot.Builder.Dialogs/DialogContainer.cs

```diff
@@ -69,8 +69,7 @@ protected virtual async Task CheckForVersionChangeAsync(DialogContext dc, Cancel
                 var handled = await dc.EmitEventAsync(DialogEvents.VersionChanged, this.Id, true, false, cancellationToken).ConfigureAwait(false);
                 if (!handled)
                 {
-                    // Throw an error for bot to catch
-                    throw new Exception($"Version change detected for '{this.Id}' dialog.");
+                    System.Diagnostics.Trace.TraceWarning($"Version change detected for '{this.Id}' dialog and wasn't handled. You should add event handler to restart dialogs when this happens.");
```

Comment:
This seems like a reasonable thing to do now, in order to unblock people. 

I wonder what is the best way longer term. Most people won't browse app insights or the trace output to find this unless there is a symptom in their experience generally. We could have this as a default behavior and add docs on how to handle this event with a good, relevant example as a first step.

Reply:
Could we call `Dialogs.TelemetryClient.TrackEvent()` instead? This would be better behavior in JS as well, as otherwise we're forced to log to the console.

Reply:
I removed this